adev/shared-docs/pipeline/guides/extensions/docs-code/docs-code.ts

```diff
@@ -18,8 +18,9 @@ export interface DocsCodeToken extends CodeToken {
 
 // Capture group 1: all attributes on the opening tag
 // Capture group 2: all content between the open and close tags
-const singleFileSelfClosingCodeRule = /^\s*<docs-code\s([^>]*)((?:.(?!\/>))*)\/>/s;
-const singleFileCodeRule = /^\s*<docs-code\s([^>]*)>((?:.(?!\/docs-code))*)<\/docs-code>/s;
+const singleFileSelfClosingCodeRule = /^\s*<docs-code\s*(.*?)\s*(?=\/>)\/>\s*$/;
+const singleFileCodeRule =
+  /^\s*<docs-code((?:\s+[\w-]+(?:="[^"]*"|='[^']*'|=\S*)>?)*)\s*(?:\/>|(.*?)<\/docs-code>)/s;
```

Comment:
A couple of corrections:

1. With the new regex, you no longer need `singleFileSelfClosingCodeRule`. Just one regex is enough, since it can capture both forms (regular and self-closing).
2. Some fixes/improvements for the regex:
    ```suggestion
      /^\s*<docs-code((?:\s+[\w-]+(?:="[^"]*"|='[^']*'|=[^\s>]*)?)*)\s*(?:\/>|>(.*?)<\/docs-code>)/s;
    ```

Reply:
Done changes verified locally